exposure time different in devdeps regtest runs

There are a few failures in the most recent devdeps regtest run:
https://github.com/spacetelescope/RegressionTests/actions/runs/12422969700/job/34685745481
```
E       AssertionError: 
E         Diff report for:
E             result file: /runner/_work/_temp/pytest_basetemp/popen-gw0/test_mos_pipeline_rtdata_module0/r0099101001001001001_F158_visit_coadd.asdf
E                 model type: MosaicModel
E             truth file: /runner/_work/_temp/pytest_basetemp/popen-gw0/test_mos_pipeline_rtdata_module0/truth/r0099101001001001001_F158_visit_coadd.asdf
E                 model type: MosaicModel
E         
E         {'values_changed': {"root['roman']['meta']['basic']['mean_exposure_time']": {'new_value': 304.96685791015625,
E                                                                                      'old_value': 304.96923828125}}}
```
Which show a difference in the computed mean exposure time.

These differences do not appear in the normal regtest run. Dependency differences are as follows
```diff
< asdf @ git+https://github.com/asdf-format/asdf.git@dd84c660feb25b46726e48086d34503733d08915  # via asdf-astropy, asdf-coordinates-schemas, crds, gwcs, rad, roman-datamodels, romancal, stcal, stpipe, stpreview, -r requirements-dev-thirdparty.txt
< asdf-astropy @ git+https://github.com/astropy/asdf-astropy.git@8147b824d56ab24a5c3f7572e43215c14fe7c134  # via gwcs, rad, roman-datamodels, romancal, -r requirements-dev-thirdparty.txt
< asdf-coordinates-schemas @ git+https://github.com/asdf-format/asdf-coordinates-schemas.git@b04d5ee1f2971545fab2a1f7a75314f9a8d48682  # via asdf-astropy, asdf-wcs-schemas, -r requirements-dev-thirdparty.txt
< asdf-standard @ git+https://github.com/asdf-format/asdf-standard.git@c17c3ecabcad7b65e41e0fb79054988b37f45a38  # via asdf, asdf-astropy, asdf-coordinates-schemas, asdf-transform-schemas, asdf-wcs-schemas, roman-datamodels, -r requirements-dev-thirdparty.txt
< asdf-transform-schemas @ git+https://github.com/asdf-format/asdf-transform-schemas.git@d49b9f5698484d02fca1bb349d0424142189f4f5  # via asdf, asdf-astropy, asdf-wcs-schemas, -r requirements-dev-thirdparty.txt
< asdf-wcs-schemas @ git+https://github.com/asdf-format/asdf-wcs-schemas.git@7b67f8d55fa0233c90335ae4f835394a519b90ce  # via gwcs, -r requirements-dev-thirdparty.txt
< astropy==7.1.dev313+g93b6f551b9  # via asdf-astropy, astroquery, crds, drizzle, gwcs, photutils, poppy, pysiaf, pyvo, roman-datamodels, romancal, spherical-geometry, stcal, stpipe, stpreview, synphot, tweakwcs, webbpsf, -r requirements-dev-thirdparty.txt
---
> asdf==4.0.0               # via asdf-astropy, asdf-coordinates-schemas, crds, gwcs, rad, roman-datamodels, romancal, stcal, stpipe, stpreview
> asdf-astropy==0.7.0       # via gwcs, rad, roman-datamodels, romancal
> asdf-coordinates-schemas==0.3.0  # via asdf-astropy, asdf-wcs-schemas
> asdf-standard==1.1.1      # via asdf, asdf-astropy, asdf-coordinates-schemas, asdf-transform-schemas, asdf-wcs-schemas, roman-datamodels
> asdf-transform-schemas==0.5.0  # via asdf, asdf-astropy, asdf-wcs-schemas
> asdf-wcs-schemas==0.4.0   # via gwcs
> astropy==7.0.0            # via asdf-astropy, astroquery, crds, drizzle, gwcs, photutils, poppy, pysiaf, pyvo, roman-datamodels, romancal, spherical-geometry, stcal, stpipe, stpreview, synphot, tweakwcs, webbpsf
12,15c12,14
< pluggy==1.5.0             # via pytest
< poppy @ git+https://github.com/spacetelescope/poppy.git@b058985f9fcdb2f64f37f3f6b703c869e5b47187  # via webbpsf, -r requirements-dev-st.txt
< pycparser==2.22           # via cffi
< pyerfa==2.0.1.5           # via astropy
---
> beautifulsoup4==4.12.3    # via astroquery
> certifi==2024.12.14       # via requests
> cffi==1.17.1              # via cryptography
31c30
< rad @ git+https://github.com/spacetelescope/rad.git@8d06d03023b9df2b9db7e19379959f30cbcbbaf7  # via roman-datamodels, -r requirements-dev-st.txt
---
> rad @ git+https://github.com/spacetelescope/rad.git@8d06d03023b9df2b9db7e19379959f30cbcbbaf7  # via roman-datamodels
36c35
< roman-datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git@2f6c46f309b642eebc11b7bafe77f8e218e5520b  # via romancal, -r requirements-dev-st.txt
---
> roman-datamodels @ git+https://github.com/spacetelescope/roman_datamodels.git@2f6c46f309b642eebc11b7bafe77f8e218e5520b  # via romancal
40c39
< scipy==1.16.0.dev0        # via gwcs, photutils, poppy, pysiaf, romancal, scikit-image, stcal, synphot, webbpsf, -r requirements-dev-thirdparty.txt
---
> scipy==1.14.1             # via gwcs, photutils, poppy, pysiaf, romancal, scikit-image, stcal, synphot, webbpsf
48,49c47,48
< stcal @ git+https://github.com/spacetelescope/stcal.git@495338006fa6faea23e8de3ddd8a506534331b98  # via romancal, -r requirements-dev-st.txt
< stpipe @ git+https://github.com/spacetelescope/stpipe.git@9fcd28daed87ad34f272da30efa21bf9ca554ae5  # via romancal, -r requirements-dev-st.txt
---
> stcal @ git+https://github.com/spacetelescope/stcal.git@495338006fa6faea23e8de3ddd8a506534331b98  # via romancal
> stpipe @ git+https://github.com/spacetelescope/stpipe.git@9fcd28daed87ad34f272da30efa21bf9ca554ae5  # via romancal
55c54
< tweakwcs @ git+https://github.com/spacetelescope/tweakwcs.git@5c187de3f7425c3403dffa44bb538e99361bcaa4  # via romancal, stcal, -r requirements-dev-st.txt
---
> tweakwcs==0.8.9           # via romancal, stcal
59c58
< webbpsf @ git+https://github.com/spacetelescope/webbpsf.git@207f724a881a224225f2e57243cefa09ffe4c04b  # via romancal, -r requirements-dev-st.txt
---
> webbpsf==1.5.0            # via romancal
```

--- a/romancal/resample/resample.py
+++ b/romancal/resample/resample.py
@@ -239,7 +239,9 @@ class ResampleData(Resample):
         if self.compute_exptime and hasattr(self, "_exptime_resampler"):
             exptime_total = self._exptime_resampler.finalize()
             m = exptime_total > 0
-            total_exposure_time = np.mean(exptime_total[m]) if np.any(m) else 0
+            total_exposure_time = (
+                np.mean(exptime_total[m], dtype="f8") if np.any(m) else 0
+            )
             max_exposure_time = np.max(exptime_total)
             log.info(
                 f"Mean, max exposure times: {total_exposure_time:.1f}, "

